patches/tModLoader/Terraria/ModLoader/NPCLoader.cs

```diff
@@ -370,7 +370,9 @@ public static byte[] WriteExtraAI(NPC npc)
 		bitWriter.Flush(modWriter);
 		modWriter.Write(bufferedStream.ToArray());
 
-		return stream.ToArray();
+		byte[] bytes = stream.ToArray();
+		// If the only byte is the bitWriter.Flush length byte, no extra data.
+		return bytes.Length == 1 && bytes[0] == 0 ? null : bytes;
```

Comment:
Kinda feels like the `bytes[0] == 0` should be an assert when `bytes.Length == 1`